pandas/core/indexes/base.py

```diff
@@ -2256,15 +2256,15 @@ def intersection(self, other):
             indexer = indexer.take((indexer != -1).nonzero()[0])
         except:
             # duplicates
-            indexer = Index(other._values).get_indexer_non_unique(
-                self._values)[0].unique()
+            indexer = algos.unique1d(Index(other._values).get_indexer_non_unique(
```

Comment:
But this whole change is to make get_indexer_non_unique return a ndarray rather than an Index so can no longer call this `Index.unique()` method.